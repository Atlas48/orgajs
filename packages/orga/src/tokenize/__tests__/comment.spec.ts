import tok from "./tok"

describe("tokenize comment", () => {
  it("knows comments", () => {
    expect(tok("# a comment")).toMatchInlineSnapshot(`
      Array [
        Object {
          "_text": "# a comment",
          "type": "comment",
          "value": "a comment",
        },
      ]
    `)
    expect(tok("# ")).toMatchInlineSnapshot(`
      Array [
        Object {
          "_text": "# ",
          "type": "comment",
          "value": "",
        },
      ]
    `)
    expect(tok("# a commentðŸ˜¯")).toMatchInlineSnapshot(`
      Array [
        Object {
          "_text": "# a commentðŸ˜¯",
          "type": "comment",
          "value": "a commentðŸ˜¯",
        },
      ]
    `)
    expect(tok(" # a comment")).toMatchInlineSnapshot(`
      Array [
        Object {
          "_text": "# a comment",
          "type": "comment",
          "value": "a comment",
        },
      ]
    `)
    expect(tok("  \t  # a comment")).toMatchInlineSnapshot(`
      Array [
        Object {
          "_text": "# a comment",
          "type": "comment",
          "value": "a comment",
        },
      ]
    `)
    expect(tok("#   a comment")).toMatchInlineSnapshot(`
      Array [
        Object {
          "_text": "#   a comment",
          "type": "comment",
          "value": "a comment",
        },
      ]
    `)
    expect(tok("#    \t a comment")).toMatchInlineSnapshot(`
      Array [
        Object {
          "_text": "#    	 a comment",
          "type": "comment",
          "value": "a comment",
        },
      ]
    `)
  })

  it("knows these are not comments", () => {
    expect(tok("#not a comment")).toMatchInlineSnapshot(`
      Array [
        Object {
          "_text": "#not a comment",
          "type": "text.plain",
          "value": "#not a comment",
        },
      ]
    `)
    expect(tok("  #not a comment")).toMatchInlineSnapshot(`
      Array [
        Object {
          "_text": "#not a comment",
          "type": "text.plain",
          "value": "#not a comment",
        },
      ]
    `)
  })
})
