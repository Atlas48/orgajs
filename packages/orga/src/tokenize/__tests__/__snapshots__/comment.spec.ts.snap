// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tokenize comment knows comments 1`] = `
Array [
  Object {
    "position": Object {
      "end": Object {
        "column": 11,
        "line": 0,
      },
      "start": Object {
        "column": 0,
        "line": 0,
      },
    },
    "type": "comment",
    "value": "",
  },
]
`;

exports[`tokenize comment knows comments 2`] = `
Array [
  Object {
    "position": Object {
      "end": Object {
        "column": 2,
        "line": 0,
      },
      "start": Object {
        "column": 0,
        "line": 0,
      },
    },
    "type": "comment",
    "value": "",
  },
]
`;

exports[`tokenize comment knows comments 3`] = `
Array [
  Object {
    "position": Object {
      "end": Object {
        "column": 13,
        "line": 0,
      },
      "start": Object {
        "column": 0,
        "line": 0,
      },
    },
    "type": "comment",
    "value": "",
  },
]
`;

exports[`tokenize comment knows comments 4`] = `
Array [
  Object {
    "position": Object {
      "end": Object {
        "column": 12,
        "line": 0,
      },
      "start": Object {
        "column": 1,
        "line": 0,
      },
    },
    "type": "comment",
    "value": "",
  },
]
`;

exports[`tokenize comment knows comments 5`] = `
Array [
  Object {
    "position": Object {
      "end": Object {
        "column": 16,
        "line": 0,
      },
      "start": Object {
        "column": 5,
        "line": 0,
      },
    },
    "type": "comment",
    "value": "",
  },
]
`;

exports[`tokenize comment knows comments 6`] = `
Array [
  Object {
    "position": Object {
      "end": Object {
        "column": 13,
        "line": 0,
      },
      "start": Object {
        "column": 0,
        "line": 0,
      },
    },
    "type": "comment",
    "value": "",
  },
]
`;

exports[`tokenize comment knows comments 7`] = `
Array [
  Object {
    "position": Object {
      "end": Object {
        "column": 16,
        "line": 0,
      },
      "start": Object {
        "column": 0,
        "line": 0,
      },
    },
    "type": "comment",
    "value": "",
  },
]
`;

exports[`tokenize comment knows these are not comments 1`] = `
Array [
  Object {
    "position": Object {
      "end": Object {
        "column": 14,
        "line": 0,
      },
      "start": Object {
        "column": 0,
        "line": 0,
      },
    },
    "type": "text.plain",
  },
]
`;

exports[`tokenize comment knows these are not comments 2`] = `
Array [
  Object {
    "position": Object {
      "end": Object {
        "column": 16,
        "line": 0,
      },
      "start": Object {
        "column": 2,
        "line": 0,
      },
    },
    "type": "text.plain",
  },
]
`;
